### PR TITLE
Adds particles to sleeping mobs (zzz)

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
https://user-images.githubusercontent.com/66640614/228422600-0696defc-eb11-46e9-bd84-1a2500cded34.mp4

Adds floating particles to sleeping humanoids.

Snore emoting will also give you these particles temporarily!

## Why It's Good For The Game
It's a cute visual that communicates a status condition

## Changelog
:cl: Tattle
add: Sleeping mobs now have zzzs when they take a nap
/:cl:

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
honestly wish there were more Z's. Sleeping crew are rare enough that it might warrent it, plus makes it easier to notice at a glance..?

**Comment by Pumpkinoe** [comment]
This is totally snoozepilled, this will help me sleepmax

**Comment by timothymtorres** [comment]
Any chance you could use matrix transforms to enlarge the sprite slowly over time, so we get a small z that grows into a big Z.  Might be asking for too much though since what you have is already really cool.

**Comment by kawoppi** [comment]
don't let the bedbugs bite!

**Comment by Shadowflame909** [comment]
i hope sleep and unconscious arent different things or it may be harder to explain to sec why im dragging around the clown with no sleep particles

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/status_effects/debuffs/debuffs.dm [hunk 1]
```diff
@@ -134,12 +134,15 @@
 		tick_interval = -1
 	RegisterSignal(owner, SIGNAL_ADDTRAIT(TRAIT_SLEEPIMMUNE), PROC_REF(on_owner_insomniac))
 	RegisterSignal(owner, SIGNAL_REMOVETRAIT(TRAIT_SLEEPIMMUNE), PROC_REF(on_owner_sleepy))
+	owner.particles = new /particles/sleeping_zs
+	owner.particles.position = list(8, 12, ABOVE_MOB_LAYER)
```
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/datums/status_effects/debuffs/debuffs.dm:138]: don't we already have a particle holder

**Comment by LemonInTheDark** [comment]
I love this so much, thank you

**Comment by zxaber** [comment]
Do these show on dead bodies to clowns?

File: code/datums/status_effects/debuffs/debuffs.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/datums/status_effects/debuffs/debuffs.dm:138]: Oh I see, sorry for misunderstanding you. 
Why a component in particular? Seems like it'd be hard to give removal commands to.
My gut says lazylist on atom of typepath -> particle but I may just be being old again

**Comment by Wallemations** [comment]
> Any chance you could use matrix transforms to enlarge the sprite slowly over time, so we get a small z that grows into a big Z.  Might be asking for too much though since what you have is already really cool.

I'd think if you could figure out the system used for smoke from ovens, then you could easily replace the smoke texture with some Z's and fiddle with the numbers to make it work!

**Comment by Time-Green** [comment]
> > Any chance you could use matrix transforms to enlarge the sprite slowly over time, so we get a small z that grows into a big Z.  Might be asking for too much though since what you have is already really cool.
> 
> I'd think if you could figure out the system used for smoke from ovens, then you could easily replace the smoke texture with some Z's and fiddle with the numbers to make it work!

There’s a growth variable on the particles, it’s more of a design decision

**Comment by Striders13** [comment]
fake death nerf

**Comment by ChungusGamer666** [comment]
as cute as this is, it has a bunch of weird implications to gameplay and i am not sure how i feel about that

**Pull request closed without merging**